OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -142,6 +142,19 @@ void INotifyCreated.Created(Actor self)
 		{
 			aircraft = self.TraitOrDefault<Aircraft>();
 			conditionManager = self.TraitOrDefault<ConditionManager>();
+
+			if (cargo.Any())
```

Comment:
You can pull the `conditionManager != null` check up here and thus save enumerating all cargo if none is present.